data/lua/explode_ability.lua

```diff
@@ -0,0 +1,38 @@
+
+local _ = wesnoth.textdomain 'wesnoth-help'
+local T = wml.tag
+
+-- The explode event code
+wesnoth.game_events.add_repeating("die", function()
+	local ec = wesnoth.current.event_context
+
+	if not ec.x1 or not ec.y1 then
+		return
+	end
+
+	local u_exploder = wesnoth.units.get(ec.x1, ec.y1)
+
+	if not u_exploder or not u_exploder:matches { ability = "explode" } then
```

Comment:
I'm trying to think of a use case where you'd want to trigger this ability in a way that uses different keys and can stack with the existing ability, but can't come up with anything.  Am I missing something?

Reply:
I guess my main reasoning is that the way you've defined it makes it more like a class of abilities rather than a single ability, and for built-in abilities, that's what the tag name indicates.